syntax = "proto3";

package noteanlyzer;

option go_package = "internal/proto/note-analyzer";

service NoteAnalyzer {
    rpc Analyze (AudioRequest) returns (NoteResponse);
}

message AudioRequest {
    AudioFileData audio_data = 1;
}

message NoteResponse {
    repeated NoteEvent notes = 1;
}

message AudioFileData {
    string file_name = 1;
    bytes audio_bytes = 2;
}

message NoteEvent {
    float start_seconds = 1;  // Время начала ноты
    int32 midi_pitch = 2;     // Например, 40 = E2, 64 = E4 и т.д.
    float velocity = 3;       // Громкость/интенсивность (если доступно)
    float duration_seconds = 4;
}